src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -312,6 +312,37 @@
     (future-cancel command-loader))
   context)
 
+;; A map of required postgres settings and their expected value.
+;; PuppetDB will refuse to start unless ALL of these values
+;; match the actual database configuration settings.
+(def required-pg-settings
+  {:standard_conforming_strings "on"})
+
+(defn verify-database-setting
+  "Given m, a map of database settings, verify 'expected', which has the form
+  [setting-name expected-value]."
+  [m expected]
+  (let [setting (first expected)
+        value (second expected)
+        map-path [setting :setting]]
+    (when-not (= (get-in m map-path) value)
+      {setting {:expected value :actual (get-in m map-path)}})))
+
+(defn verify-database-settings
+  "Ensure the database configuration does not have any settings known
+  to break PuppetDB. If an invalid database configuration is found, throws
+  {:type ::invalid-database-configuration :failed-validation failed-map}"
+  [db-conn-pool]
+  (let [database-settings (jdbc/with-db-connection db-conn-pool (jdbc/query "show all;" ))
+        munged-settings (apply merge
+                                (map (fn [h]
+                                       {(keyword (:name h)) (dissoc h :name)})
+                                     database-settings))
+        invalid-settings (doall (map #(verify-database-setting munged-settings %) required-pg-settings))]
+    (if (seq (filter some? invalid-settings))
```

Comment:
I'd suggest a `when` here -- not critical, but it can help make it clear that this is intentionally "one-armed", and could also be `(remove nil? ...`.  I'd thought about suggesting mapv instead of `(doall (map ...` since it's also eager, but I suspect you don't need either, i.e. if it matters at all, the the merge will force the issue below.